### PR TITLE
UIButton: Add choice_input / dropdown selections

### PR DESCRIPTION
**Pull request by jbadlato**
Implements #5898 

I have pretty limited experience with typescript/react/go, so I'm sure there's some convention issues/bad practices here (including some `@ts-ignore`s that I don't know how/if to fix.  I cargo-culted pretty much everything here from #4893 & what else I could gather from the surrounding code.

I also considered implementing this as radio buttons, but that should probably be easy to expose as another parameter to `choice_input` as a future improvement?

choice_input

Smoke tested this by editing the oneup Tiltfile: 
```
diff --git a/integration/oneup/Tiltfile b/integration/oneup/Tiltfile
index 1cf5dda00..62b0e1c71 100644
--- a/integration/oneup/Tiltfile
+++ b/integration/oneup/Tiltfile
@@ -1,5 +1,6 @@
 # -*- mode: Python -*-
 
+load('ext://uibutton', 'cmd_button')
 include('../Tiltfile')
 
 # If you get push errors, you can change the default_registry.
@@ -11,3 +12,20 @@ default_registry(read_json('tilt_option.json', {})
 docker_build('oneup', '.')
 k8s_yaml('oneup.yaml')
 k8s_resource('oneup', port_forwards=8100)
+
+
+def choice_input(name, label='', default='', choices=[]):
+    result = {
+        "name": name,
+        "label": label,
+        "choice": {
+            "defaultValue": default,
+            "choices": choices
+        }
+    }
+    return result
+
+cmd_button(name='Test Choice',
+           resource='oneup',
+           argv=['sh', '-c', "echo choice: $MY_CHOICE_INPUT"],
+           inputs=[choice_input(name='MY_CHOICE_INPUT', label='input name label', default='choice2', choices=['choice1', 'choice2', 'choice3'])])
```

### CODE REVIEW

File: pkg/apis/core/v1alpha1/uibutton_types.go [hunk 1]
```diff
@@ -253,6 +253,23 @@ type UIHiddenInputStatus struct {
 	Value string `json:"value" protobuf:"bytes,1,opt,name=value"`
 }
 
+// Describes a choice dropdown input field attached to a button
+type UIChoiceInputSpec struct {
+	// Initial value for this field.
+	//
+	// +optional
+	DefaultValue string `json:"defaultValue" protobuf:"bytes,1,opt,name=defaultValue"`
```
Comment (nicks) [inline pkg/apis/core/v1alpha1/uibutton_types.go:261]: is it ok to have a DefaultValue that's not an element of the Choices array?

alternatively, what do you think of getting rid of DefaultValue, and making the default value implicitly the first Choice?

File: pkg/apis/core/v1alpha1/uibutton_types.go [hunk 2]
```diff
@@ -292,6 +312,9 @@ func (in *UIInputSpec) Validate(_ context.Context, path *field.Path) field.Error
 	if in.Hidden != nil {
 		numInputTypes += 1
 	}
+	if in.Choice != nil {
+		numInputTypes += 1
```
Comment (nicks) [inline pkg/apis/core/v1alpha1/uibutton_types.go:316]: we should also validate that Choices is non-empty if this field is defined

File: web/tilt_modules/uibutton/README.md [hunk 3]
```diff
@@ -54,6 +54,17 @@ checkbox's state. By default, the variable will be set to the string `"true"` or
 | `true_string`  | `str`  | If not None, when the checkbox is checked, the environment variable will be set to this string instead of "true". |
 | `false_string` | `str`  | If not None, when the checkbox is checked, the environment variable will be set to this string instead of "false". |
 
+### `choice_input(name, label='', default='', choices=[])`
```
Comment (nicks) [inline web/tilt_modules/uibutton/README.md:57]: i think this tilt_modules directory is vestigial, lemme remove it in another pr

**Review by nicks — COMMENTED** [review]
nice!! quick high-level question about the schema

File: pkg/apis/core/v1alpha1/uibutton_types.go (hunk 1 above)
Comment (jbadlato) [reply pkg/apis/core/v1alpha1/uibutton_types.go:261]: Yeah now that you mention it, that seems like a way better solution.  I'll get rid of the DefaultValue

File: web/tilt_modules/uibutton/README.md (hunk 3 above)
Comment (jbadlato) [reply web/tilt_modules/uibutton/README.md:57]: Should I be making a separate PR to https://github.com/tilt-dev/tilt-extensions/?

Comment (nicks) [reply web/tilt_modules/uibutton/README.md:57]: ya, but you'll have to wait for a release

Comment (jbadlato) [reply web/tilt_modules/uibutton/README.md]: Sounds good!

**Comment by jbadlato** [comment]
@nicks Alright I think I've cleaned up all the code with the suggestions so far.  I'm not sure how to resolve the DCO check, since the details only give remedy instructions if I don't have empty commits on my branch, which I do.  Should I just squash these to a new branch and create a new PR?

**Comment by nicks** [comment]
@jbadlato thanks! ya, it's probably easiest to squash and push --force the pr

**Review by nicks — APPROVED** [review]
lgtm!

**Pull request merged**